server/src/main/java/org/opensearch/common/util/FeatureFlags.java

```diff
@@ -37,6 +37,13 @@ public class FeatureFlags {
      */
     public static final String SEARCHABLE_SNAPSHOT = "opensearch.experimental.feature.searchable_snapshot.enabled";
 
+    /**
+     * Gates the ability for Searchable Snapshots to read snapshots that are older than the
+     * guaranteed backward compatibility for OpenSearch (one prior major version) on a best effort basis.
+     */
+    public static final String SEARCHABLE_SNAPSHOT_EXTENDED_COMPATIBILITY =
```

Comment:
This might be a bad idea, but can we consider not adding another feature flag for this? If we need to promote searchable snapshots out of experimental without the extended compatibility support, we can put a new feature flag in at that time.

Reply:
I'm ok with doing this. I took the more conservative route to protect against a contributor removing the searchable snapshots feature flag before the backwards compatibility changes were sufficiently tested.